packages/Search/src/details/DTK_DetailsTreeVisualization.hpp

```diff
@@ -0,0 +1,144 @@
+        auto const node_index = TreeAccess::getIndex( node, tree );
+        std::string label = node_is_leaf ? "l" : "i";
+        label.append( std::to_string( node_index ) );
+        return label;
+    }
+
+    template <typename Tree>
+    static std::string getNodeAttributes( Node const *node, Tree const &tree )
+    {
+        return TreeAccess::isLeaf( node, tree ) ? "[leaf]" : "[internal]";
+    }
+
+    template <typename Tree>
+    static std::string getEdgeAttributes( Node const *parent, Node const *child,
+                                          Tree const &tree )
+    {
+        return TreeAccess::isLeaf( child, tree ) ? "[pendant]" : "[edge]";
+    }
+
+    struct GraphvizVisitor
+    {
+        std::ostream &_os;
+
+        template <typename Tree>
+        void visit( Node const *node, Tree const &tree ) const
+        {
+            visitNode( node, tree );
+            visitEdgesStartingFromNode( node, tree );
+        }
+
+        template <typename Tree>
+        void visitNode( Node const *node, Tree const &tree ) const
+        {
+            auto const node_label = getNodeLabel( node, tree );
+            auto const node_attributes = getNodeAttributes( node, tree );
+
+            _os << "    " << node_label << " " << node_attributes << ";\n";
+        }
+
+        template <typename Tree>
+        void visitEdgesStartingFromNode( Node const *node,
+                                         Tree const &tree ) const
+        {
+            auto const node_label = getNodeLabel( node, tree );
+            auto const node_is_internal = !TreeAccess::isLeaf( node, tree );
+
+            if ( node_is_internal )
+                for ( Node const *child :
+                      {node->children.first, node->children.second} )
+                {
+                    auto const child_label = getNodeLabel( child, tree );
+                    auto const edge_attributes =
+                        getEdgeAttributes( node, child, tree );
+
+                    _os << "    " << node_label << " -> " << child_label << " "
+                        << edge_attributes << ";\n";
+                }
+        }
+    };
+
+    template <typename Tree, typename Visitor>
+    static void visitAllIterative( Tree const &tree, Visitor const &visitor )
+    {
+        Stack<Node const *> stack;
+        stack.emplace( TreeAccess::getRoot( tree ) );
+        while ( !stack.empty() )
+        {
+            Node const *node = stack.top();
+            stack.pop();
+
+            visitor.visit( node, tree );
+
+            if ( !TreeAccess::isLeaf( node, tree ) )
+                for ( Node const *child :
+                      {node->children.first, node->children.second} )
+                    stack.push( child );
+        }
+    }
+
+    template <typename Tree, typename Predicate, typename Visitor>
+    static int visit( Tree const &tree, Predicate const &pred,
+                      Visitor const &visitor )
+    {
+#if defined( __CUDA_ARCH__ )
```

Comment:
You need to slow down on the macro. You can can check the `DeviceType`

Reply:
This is the only way I could think of to suppress the warning with nvcc.  What other solution do you propose?

Reply:
Added a comment to capture this.